scripts/spack/packages/axom/package.py

```diff
@@ -136,18 +139,35 @@ class Axom(CachedCMakePackage, CudaPackage, ROCmPackage):
         depends_on("raja@:0.13.0", when="@:0.5.0")
         depends_on("raja~openmp", when="~openmp")
         depends_on("raja+openmp", when="+openmp")
+        depends_on("raja+cuda", when="+cuda")
+
+    with when("+profiling"):
+        depends_on("adiak")
+        depends_on("caliper+adiak~papi")
+
+        depends_on("caliper+cuda", when="+cuda")
+        depends_on("caliper~cuda", when="~cuda")
+
+        for dep in ["adiak", "caliper"]:
+            depends_on("{0}+mpi".format(dep), when="+mpi")
+            depends_on("{0}~mpi".format(dep), when="~mpi")
+            depends_on("{0}+shared".format(dep), when="+shared")
+            depends_on("{0}~shared".format(dep), when="~shared")
+
 
     for val in CudaPackage.cuda_arch_values:
         raja_cuda = "raja +cuda cuda_arch={0}".format(val)
         umpire_cuda = "umpire +cuda cuda_arch={0}".format(val)
         depends_on(raja_cuda, when="+{0}".format(raja_cuda))
         depends_on(umpire_cuda, when="+{0}".format(umpire_cuda))
+        depends_on("caliper cuda_arch={0}".format(val), when="+profiling cuda_arch={0}".format(val))
 
     for val in ROCmPackage.amdgpu_targets:
         raja_rocm = "raja +rocm amdgpu_target={0}".format(val)
         umpire_rocm = "umpire +rocm amdgpu_target={0}".format(val)
         depends_on(raja_rocm, when="+{0}".format(raja_rocm))
         depends_on(umpire_rocm, when="+{0}".format(umpire_rocm))
+        depends_on("caliper amdgpu_target={0}".format(val), when="+profiling amdgpu_target={0}".format(val))
```

Comment:
`+rocm` is not getting set for this target. Is this correct? Also it would be nice to follow the same pattern for all of these. The `+cuda` got mixed as well. The for loop for the `cuda_arch` and `amdgpu_target` handles `cuda` and  `rocm` respectively. This was probably changed since this was originally written.

Reply:
Thanks -- I updated the recipe and am rebuilding the rocm and cuda TPLs